server/sonar-server/src/main/java/org/sonar/server/ws/RequestVerifier.java

```diff
@@ -23,17 +23,25 @@
 import org.sonar.api.server.ws.WebService;
 import org.sonar.server.exceptions.ServerException;
 
-import javax.servlet.http.HttpServletResponse;
+import static javax.servlet.http.HttpServletResponse.SC_METHOD_NOT_ALLOWED;
 
 public class RequestVerifier {
   private RequestVerifier() {
     // static methods only
   }
 
   public static void verifyRequest(WebService.Action action, Request request) {
-    // verify the HTTP verb
-    if (action.isPost() && !"POST".equals(request.method())) {
-      throw new ServerException(HttpServletResponse.SC_METHOD_NOT_ALLOWED, "HTTP method POST is required");
+    switch (request.method()) {
+      case "GET":
+        if (action.isPost()) {
+          throw new ServerException(SC_METHOD_NOT_ALLOWED, "HTTP method POST is required");
+        }
+        return;
+      case "PUT":
+      case "DELETE":
+        throw new ServerException(SC_METHOD_NOT_ALLOWED, String.format("HTTP method %s is not allowed", request.method()));
```

Comment:
PUT and DELETE are not used in Java WS => The goal here is to explain to user that these 2 methods are not allowed anymore.